drivers/sensor/st/lsm9ds1/lsm9ds1.c

```diff
@@ -0,0 +1,677 @@
+/*
+ * Copyright (c) 2024 Bootlin
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_lsm9ds1
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/logging/log.h>
+
+#include "lsm9ds1.h"
+#include <stmemsc.h>
+
+LOG_MODULE_REGISTER(LSM9DS1, CONFIG_SENSOR_LOG_LEVEL);
+
+/* Sensitivity of the accelerometer, indexed by the raw full scal value. Unit is µg/ LSB */
+static const uint16_t lsm9ds1_accel_fs_sens[] = {61, 732, 122, 244};
+
+/*
+ * Sensitivity of the gyroscope, indexed by the raw full scale value.
+ * The value here is just a factor applied to GAIN_UNIT_G, as the sensitivity is
+ * proportional to the full scale size.
+ */
+static const uint16_t lsm9ds1_gyro_fs_sens[] = {1, 2, 0, 8};
```

Comment:
Should the `0` be `4`?

Reply:
The raw odr value (index of the array) can only be 0,1 or 3. I added a comment to explain it, but maybe I should put another value than `0` ?